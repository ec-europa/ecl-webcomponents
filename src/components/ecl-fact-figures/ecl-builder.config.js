const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, './node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-fact-figures-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-fact-figures-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-fact-figures-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-fact-figures-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-fact-figures-item-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-fact-figures-item-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-fact-figures-item-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-fact-figures-item-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
