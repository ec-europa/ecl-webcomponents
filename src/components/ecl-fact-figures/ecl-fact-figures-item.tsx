import { Component, h, Prop} from '@stencil/core';

@Component({
  tag: 'ecl-fact-figures-item',
  shadow: false,
  styleUrls: {
    ec: './build/styles/ecl-fact-figures-item-ec.css',
    eu: './build/styles/ecl-fact-figures-item-eu.css',
  },
  scoped: true,
})
export class EclFactFiguresItem {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() icon: string;
  @Prop() value: string;
  @Prop() itemTitle: string;
  @Prop() iconTransform: string;

  getClass(): string {
    return [
      `ecl-fact-figures__item`,
      `sc-ecl-fact-figures-${this.theme}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <div class={this.getClass()}>
        <ecl-icon
          icon={this.icon}
          size={this.theme === 'ec' ? 'm' : 'l'}
          transform={this.iconTransform}
          style-class={`ecl-fact-figures__icon sc-ecl-fact-figures-item-${this.theme}`}
        ></ecl-icon>
        <div class="ecl-fact-figures__value">{this.value}</div>
        <div class="ecl-fact-figures__title">{this.itemTitle}</div>
        <div class="ecl-fact-figures__description">
          <slot name="description"></slot>
        </div>
      </div>
    );
  }
}