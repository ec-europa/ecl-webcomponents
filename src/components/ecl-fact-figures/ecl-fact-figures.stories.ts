const getArgs = () => {
  return {
    columns: '3',
    centered: false,
    viewAllLinkPath: './example',
    viewAllLinkLabel: 'View all',
    fontSize: 'l',
    iconSize: 'medium',
  };
};

const getArgTypes = () => {
  return {
    columns: {
      type: { name: 'select' },
	    options: ['3', '4'],
	    table: {
	      description: "Number of columns"
	    },
	  },
    centered: {
      type: { name: 'boolean' },
    },
    viewAllLinkPath:{
      name: 'view-all-link-path',
      type: { name: 'string' },
    },
    viewAllLinkLabel: {
      name: 'view-all-link-label',
      type: { name: 'string'},
    },
    fontSize: {
      name: 'font-size',
      type: 'select',
      description: 'Change font size',
      options: ['m', 'l'],
      control: {
        labels: {
          m: 'medium',
          l: 'large',
        },
      },
      mapping: {
        medium: 'm',
        large: 'l',
      },
    },
    iconSize: {
      name: 'icon-size',
      description: 'Size of the icon',
      type: 'select',
      options: ['medium', 'large'],
      mapping: {
        medium: 'l',
        large: '2xl',
      },
    },
  };
};

export default {
  title: 'Components/fact and figures',
};

const Template = (args) => 
  `<ecl-fact-figures
    columns="${args.columns}"
    centered="${args.centered}"
    color-mode="${args.color_mode}"
    view-all-link-path="${args.viewAllLinkPath}"
    view-all-link-label="${args.viewAllLinkLabel}"
    icon-size="${args.iconSize}"
    font-size="${args.fontSize}"
  >
    <ecl-fact-figures-item
      item-title="Lorem ipsum",
      value="00.0 million"
      icon="infographic"
      theme="${args.theme}"
    >
      Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
    </ecl-fact-figures-item>

    <ecl-fact-figures-item
      item-title="Sed hendrerit",
      value="00.0 million"
      icon="spreadsheet"
      theme="${args.theme}"
    >
      Turpis varius congue venenatis, erat dui feugiat felis.
    </ecl-fact-figures-item>

    <ecl-fact-figures-item
      item-title="Donec suscipit interdum augue, ac dapibus eros finibus a.",
      value="00.0 million"
      icon="growth"
      theme="${args.theme}"
    >
      Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
    </ecl-fact-figures-item>

    <ecl-fact-figures-item
      item-title="Aenean dapibus",
      value="00.0 million"
      icon="digital"
      theme="${args.theme}"
    >
      Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
    </ecl-fact-figures-item>

    <ecl-fact-figures-item
      item-title="Aliquam faucibus nulla eget eleifend",
      value="00.0 million"
      icon="regulation"
      theme="${args.theme}"
    >
      Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
    </ecl-fact-figures-item>

    <ecl-fact-figures-item
      item-title="Aliquam faucibus nulla eget eleifend",
      value="00.0 million"
      icon="image"
      theme="${args.theme}"
    >
      Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
    </ecl-fact-figures-item>
  </ecl-fact-figures>`;

export const FactFigures = Template.bind({});
FactFigures.storyName = 'default';
FactFigures.args = getArgs();
FactFigures.argTypes = getArgTypes();
