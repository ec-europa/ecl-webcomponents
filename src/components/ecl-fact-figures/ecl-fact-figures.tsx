import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-fact-figures',
  styleUrls: {
    ec: './build/styles/ecl-fact-figures-ec.css',
    eu: './build/styles/ecl-fact-figures-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})
export class EclFactFigures {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() columns: number = 3;
  @Prop() displayIcons: boolean = true;

  getClass(): string {
    return [
      `ecl-fact-figures`,
      `ecl-fact-figures--col-${this.columns}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <div class={this.getClass()}>
        <div class="ecl-fact-figures__items">
          <slot></slot>
        </div>
      </div>
    );
  }
}
