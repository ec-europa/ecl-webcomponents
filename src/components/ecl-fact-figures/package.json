{
  "name": "@ecl/web-components-fact-figures",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.0.0",
  "description": "ECL Fact & figures",
  "main": "ecl-media-container.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.7.0",
    "@ecl/vanilla-component-fact-figures": "3.7.0",
    "@ecl/vanilla-layout-grid": "3.7.0",
    "@ecl/vanilla-utility-background": "3.7.0",
    "@ecl/vanilla-utility-border": "3.7.0",
    "@ecl/vanilla-utility-clearfix": "3.7.0",
    "@ecl/vanilla-utility-dimension": "3.7.0",
    "@ecl/vanilla-utility-disablescroll": "3.7.0",
    "@ecl/vanilla-utility-display": "3.7.0",
    "@ecl/vanilla-utility-flex": "3.7.0",
    "@ecl/vanilla-utility-float": "3.7.0",
    "@ecl/vanilla-utility-media": "3.7.0",
    "@ecl/vanilla-utility-print": "3.7.0",
    "@ecl/vanilla-utility-screen-reader": "3.7.0",
    "@ecl/vanilla-utility-spacing": "3.7.0",
    "@ecl/vanilla-utility-typography": "3.7.0",
    "@ecl/vanilla-utility-z-index": "3.7.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}