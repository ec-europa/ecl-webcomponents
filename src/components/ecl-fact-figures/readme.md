# ecl-fact-figures



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type     | Default     |
| --------------- | ---------------- | ----------- | -------- | ----------- |
| `icon`          | `icon`           |             | `string` | `undefined` |
| `iconTransform` | `icon-transform` |             | `string` | `undefined` |
| `itemTitle`     | `item-title`     |             | `string` | `undefined` |
| `styleClass`    | `style-class`    |             | `string` | `undefined` |
| `theme`         | `theme`          |             | `string` | `'ec'`      |
| `value`         | `value`          |             | `string` | `undefined` |


## Dependencies

### Depends on

- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-fact-figures-item --> ecl-icon
  style ecl-fact-figures-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
