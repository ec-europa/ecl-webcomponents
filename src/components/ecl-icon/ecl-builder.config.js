const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
