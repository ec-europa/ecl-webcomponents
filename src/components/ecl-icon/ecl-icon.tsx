import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}