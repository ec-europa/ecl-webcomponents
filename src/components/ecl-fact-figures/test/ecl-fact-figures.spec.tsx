import { newSpecPage } from '@stencil/core/testing';
import { EclFactFigures } from '../ecl-fact-figures';
import { EclFactFiguresItem } from '../ecl-fact-figures-item';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-fact-figures', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclFactFigures, EclFactFiguresItem, EclIcon],
      html: `<ecl-fact-figures
        columns="4"
        theme="ec"
      >
        <ecl-fact-figures-item
          item-title="Lorem ipsum",
          value="00.0 million"
          icon="infographic"
          theme="ec"
        >
          <span slot="description">
            Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
          </span>
        </ecl-fact-figures-item>

        <ecl-fact-figures-item
          item-title="Sed hendrerit",
          value="00.0 million"
          icon="spreadsheet"
          theme="ec"
        >
          <span slot="description">
            Turpis varius congue venenatis, erat dui feugiat felis.
          </span>
        </ecl-fact-figures-item>

        <ecl-fact-figures-item
          item-title="Donec suscipit interdum augue, ac dapibus eros finibus a.",
          value="00.0 million"
          icon="growth"
          theme="ec"
        >
          <span slot="description">
            Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
          </span>
        </ecl-fact-figures-item>

        <ecl-fact-figures-item
          item-title="Aenean dapibus",
          value="00.0 million"
          icon="digital"
          theme="ec"
        >
          <span slot="description">
            Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
          </span>
        </ecl-fact-figures-item>

        <ecl-fact-figures-item
          item-title="Aliquam faucibus nulla eget eleifend",
          value="00.0 million"
          icon="regulation"
          theme="ec"
        >
          <span slot="description">
            Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
          </span>
        </ecl-fact-figures-item>

        <ecl-fact-figures-item
          item-title="Aliquam faucibus nulla eget eleifend",
          value="00.0 million"
          icon="image"
          theme="ec"
        >
          <span slot="description">
            Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
          </span>
        </ecl-fact-figures-item>
      </ecl-fact-figures>`,
    });
    
    expect(page.root).toMatchSnapshot();
  });
});
