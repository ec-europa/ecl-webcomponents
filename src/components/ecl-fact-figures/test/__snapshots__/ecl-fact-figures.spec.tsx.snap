// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-fact-figures renders 1`] = `
<ecl-fact-figures columns="4" font-size="m" icon-size="2xl" view-all-link-label="View all" view-all-link-path="/example">
  <!---->
  <div class="ecl-fact-figures ecl-fact-figures--col-4 ecl-fact-figures--font-m">
    <div class="ecl-fact-figures__items">
      <ecl-fact-figures-item icon="infographic" item-title="Lorem ipsum" value="00.0 million">
        <!---->
        <div class="ecl-fact-figures__item ecl-fact-figures__item--font-m sc-ecl-fact-figures-ec">
          <ecl-icon style-class="ecl-fact-figures__icon sc-ecl-fact-figures-item-ec">
            <svg class="ecl-fact-figures__icon ecl-icon ecl-icon--2xl sc-ecl-fact-figures-item-ec">
              <use xlink:href="/build/images/ec/icons.svg#infographic"></use>
            </svg>
          </ecl-icon>
          <div class="ecl-fact-figures__value">
            00.0 million
          </div>
          <div class="ecl-fact-figures__title">
            Lorem ipsum
          </div>
          <div class="ecl-fact-figures__description">
            Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
          </div>
        </div>
      </ecl-fact-figures-item>
      <ecl-fact-figures-item icon="spreadsheet" item-title="Sed hendrerit" theme="ec" value="00.0 million">
        <!---->
        <div class="ecl-fact-figures__item ecl-fact-figures__item--font-m sc-ecl-fact-figures-ec">
          <ecl-icon style-class="ecl-fact-figures__icon sc-ecl-fact-figures-item-ec">
            <svg class="ecl-fact-figures__icon ecl-icon ecl-icon--2xl sc-ecl-fact-figures-item-ec">
              <use xlink:href="/build/images/ec/icons.svg#spreadsheet"></use>
            </svg>
          </ecl-icon>
          <div class="ecl-fact-figures__value">
            00.0 million
          </div>
          <div class="ecl-fact-figures__title">
            Sed hendrerit
          </div>
          <div class="ecl-fact-figures__description">
            Turpis varius congue venenatis, erat dui feugiat felis.
          </div>
        </div>
      </ecl-fact-figures-item>
      <ecl-fact-figures-item icon="growth" item-title="Donec suscipit interdum augue, ac dapibus eros finibus" value="00.0 million">
        <!---->
        <div class="ecl-fact-figures__item ecl-fact-figures__item--font-m sc-ecl-fact-figures-ec">
          <ecl-icon style-class="ecl-fact-figures__icon sc-ecl-fact-figures-item-ec">
            <svg class="ecl-fact-figures__icon ecl-icon ecl-icon--2xl sc-ecl-fact-figures-item-ec">
              <use xlink:href="/build/images/ec/icons.svg#growth"></use>
            </svg>
          </ecl-icon>
          <div class="ecl-fact-figures__value">
            00.0 million
          </div>
          <div class="ecl-fact-figures__title">
            Donec suscipit interdum augue, ac dapibus eros finibus
          </div>
          <div class="ecl-fact-figures__description">
            Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
          </div>
        </div>
      </ecl-fact-figures-item>
      <ecl-fact-figures-item icon="digital" item-title="Aenean dapibus" value="00.0 million">
        <!---->
        <div class="ecl-fact-figures__item ecl-fact-figures__item--font-m sc-ecl-fact-figures-ec">
          <ecl-icon style-class="ecl-fact-figures__icon sc-ecl-fact-figures-item-ec">
            <svg class="ecl-fact-figures__icon ecl-icon ecl-icon--2xl sc-ecl-fact-figures-item-ec">
              <use xlink:href="/build/images/ec/icons.svg#digital"></use>
            </svg>
          </ecl-icon>
          <div class="ecl-fact-figures__value">
            00.0 million
          </div>
          <div class="ecl-fact-figures__title">
            Aenean dapibus
          </div>
          <div class="ecl-fact-figures__description">
            Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
          </div>
        </div>
      </ecl-fact-figures-item>
      <ecl-fact-figures-item ,="" icon="regulation" item-title="Aliquam faucibus nulla eget eleifend" value="00.0 million">
        <!---->
        <div class="ecl-fact-figures__item ecl-fact-figures__item--font-m sc-ecl-fact-figures-ec">
          <ecl-icon style-class="ecl-fact-figures__icon sc-ecl-fact-figures-item-ec">
            <svg class="ecl-fact-figures__icon ecl-icon ecl-icon--2xl sc-ecl-fact-figures-item-ec">
              <use xlink:href="/build/images/ec/icons.svg#regulation"></use>
            </svg>
          </ecl-icon>
          <div class="ecl-fact-figures__value">
            00.0 million
          </div>
          <div class="ecl-fact-figures__title">
            Aliquam faucibus nulla eget eleifend
          </div>
          <div class="ecl-fact-figures__description">
            Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
          </div>
        </div>
      </ecl-fact-figures-item>
      <ecl-fact-figures-item icon="image" item-title="Aliquam faucibus nulla eget eleifend" value="00.0 million">
        <!---->
        <div class="ecl-fact-figures__item ecl-fact-figures__item--font-m sc-ecl-fact-figures-ec">
          <ecl-icon style-class="ecl-fact-figures__icon sc-ecl-fact-figures-item-ec">
            <svg class="ecl-fact-figures__icon ecl-icon ecl-icon--2xl sc-ecl-fact-figures-item-ec">
              <use xlink:href="/build/images/ec/icons.svg#image"></use>
            </svg>
          </ecl-icon>
          <div class="ecl-fact-figures__value">
            00.0 million
          </div>
          <div class="ecl-fact-figures__title">
            Aliquam faucibus nulla eget eleifend
          </div>
          <div class="ecl-fact-figures__description">
            Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
          </div>
        </div>
      </ecl-fact-figures-item>
    </div>
    <div class="ecl-fact-figures__view-all">
      <ecl-link style-class="ecl-fact-figures__view-all-link">
        <!---->
        <a class="ecl-fact-figures__view-all-link ecl-link ecl-link--icon ecl-link--icon-after ecl-link--standalone" href="/example">
          View all
          <slot-fb hidden="" name="icon-after"></slot-fb>
          <ecl-icon slot="icon-after">
            <svg class="ecl-icon ecl-icon--flip-horizontal ecl-icon--xs ecl-link__icon ecl-link__icon--after">
              <use xlink:href="/build/images/ec/icons.svg#arrow-left"></use>
            </svg>
          </ecl-icon>
        </a>
      </ecl-link>
    </div>
  </div>
</ecl-fact-figures>
`;
